src/tools/atlas/atlasTool.ts

```diff
@@ -1,17 +1,20 @@
 import { ToolBase } from "../tool.js";
 import { ApiClient } from "../../common/atlas/apiClient.js";
 import { State } from "../../state.js";
-import { ensureAuthenticated } from "../../common/atlas/auth.js";
 
 export abstract class AtlasToolBase extends ToolBase {
     constructor(
         state: State,
-        protected apiClient: ApiClient
+        protected apiClient?: ApiClient
     ) {
         super(state);
     }
 
     protected async ensureAuthenticated(): Promise<void> {
```

Comment:
```suggestion
    protected ensureAuthenticated(): asserts this is { apiClient: ApiClient } {
```
should let you get rid of all `!` if you remove the await-s